CriteoPublisherSdk/Sources/CR_CdbBid.m

```diff
@@ -215,13 +210,19 @@ - (BOOL)isInSilenceMode {
 }
 
 - (BOOL)isValid {
-  return [[NSScanner scannerWithString:self.cpm] scanFloat:NULL] && self.cpm.floatValue >= 0.0f &&
-         [self isValidNativeAssetsOrUrl];
+  return self.isValidCpm && self.isValidNativeAssetsOrUrl;
 }
 
 - (BOOL)isRenewable {
-  BOOL result = !self.isInSilenceMode || self.isExpired;
-  return result;
+  return !self.isInSilenceMode || self.isExpired;
+}
+
+- (BOOL)isImmediate {
```

Comment:
nit: could be a good occasion to document each of the `isXXX` methods to educate the reader on the meaning of these different modes.

Reply:
Agreed but FAR, are you thinking about more documentation than the one I've put?
> YES if a bid is "immediate". i.e. CPM > 0, TTL = 0
Also, is there any reference documentation about all these modes?

Reply:
Yes I was thinking about something more explicit that explains what `immediate` means for example. But it's really nit. There is the PubSDK overview which is pinned on the Slack channel.

Reply:
Ok will add some entry in my _personal_ todo list then :)